langchain4j-mongodb-atlas/src/test/java/dev/langchain4j/store/embedding/mongodb/TestHelper.java

```diff
@@ -0,0 +1,116 @@
+package dev.langchain4j.store.embedding.mongodb;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerApi;
+import com.mongodb.ServerApiVersion;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.model.embedding.onnx.allminilml6v2q.AllMiniLmL6V2QuantizedEmbeddingModel;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import org.bson.UuidRepresentation;
+import org.junit.jupiter.api.Assumptions;
+import org.testcontainers.mongodb.MongoDBAtlasLocalContainer;
+import org.testcontainers.shaded.com.google.common.collect.Sets;
+
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Function;
+
+import static java.lang.String.format;
+
+class TestHelper {
+
+    public static final EmbeddingModel EMBEDDING_MODEL = new AllMiniLmL6V2QuantizedEmbeddingModel();
+    private static final String USERNAME = getenv("MONGODB_ATLAS_USERNAME");
+    private static final String PASSWORD = getenv("MONGODB_ATLAS_PASSWORD");
+    private static final String HOST = getenv("MONGODB_ATLAS_HOST");
+    private static final String CONNECTION_STRING = "mongodb://localhost"; // TODO
+
+    private final String dbName;
+    private final MongoClient client;
+    private MongoDbEmbeddingStore embeddingStore;
+
+    TestHelper(MongoClient client) {
+        this.dbName = "test_database_" + ThreadLocalRandom.current().nextInt(0, Integer.MAX_VALUE);
+        this.client = client;
+    }
+
+    TestHelper initialize() {
+        return this.initialize(b -> b);
+    }
+
+    TestHelper initialize(Function<MongoDbEmbeddingStore.Builder, MongoDbEmbeddingStore.Builder> initializer) {
+        this.embeddingStore = initializer.apply(getDefaultMongoDbEmbeddingStoreBuilder()).build();
+        return this;
+    }
+
+    static void assertDoContainerTests() {
+        Assumptions.assumeTrue(getenv("MONGODB_DISABLE_CONTAINER_TESTS") == null);
+    }
+
+    EmbeddingStore<TextSegment> getEmbeddingStore() {
+        return embeddingStore;
+    }
+
+    static MongoClient createClientFromContainer(MongoDBAtlasLocalContainer mongodb) {
+        MongoClientSettings.Builder builder = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(mongodb.getConnectionString()));
+        return createClientFromBuilder(builder);
+    }
+
+    static MongoClient createClientFromEnv() {
+        boolean hasParts = HOST != null && USERNAME != null && PASSWORD != null;
+        Assumptions.assumeTrue(hasParts || CONNECTION_STRING != null,
+                "Skipping test: host/username/password, or connection string must be provided via env vars.");
+
+        String connectionString = CONNECTION_STRING != null
```

Comment:
Looks like `CONNECTION_STRING` is always non-null. How to test `MongoDB Atlas` on the cloud?

Reply:
You are right, this should be an env var. I unintentionally introduced the hardcoded value during a rebase, thank you.

Just to explain why the connection string is here at all, I used it as another option for testing with [local Atlas CLI](https://www.mongodb.com/docs/atlas/cli/current/atlas-cli-deploy-local/) rather than cloud (the recently-updated container seems just as good for local testing).

Reply:
I added this as an env var, resolved.